pkg/storage/ingest/pusher.go

```diff
@@ -24,7 +24,14 @@ import (
 	"github.com/grafana/mimir/pkg/util/spanlogger"
 )
 
-const shardForSeriesBuffer = 2000 // TODO dimitarvdimitrov 2000 is arbitrary; the idea is that we don't block the goroutine calling PushToStorage while we're flushing. A linked list with a sync.Cond or something different would also work
+// batchingQueueCapacity controls how many batches can be enqueued for flushing.
+// We want to batch and flush in parallel. There's no point in batching faster than we can flush.
+// A buffer of 1 would allow to balance the time for batching and flushing 50/50.
+// For example, if we flush 1 batch/sec, then batching 2 batches/sec doesn't make us faster.
+// If we're still flushing batch `N`, batch `N+1` is enqueued for flushing, and we've just finished batching batch `N+2`,
+// then we're going too fast - in the time to flush `N` we've created `N+2` and `N+1`.
+// A buffer of 1 doesn't let us go too far ahead with batching, but still doesn't require synchronisation between batching and flushing.
```

Comment:
```suggestion
// batchingQueueCapacity controls how many batches can be enqueued for flushing.
// We don't want to push any batches in parallel and instead want to prepare the next one while the current one finishes, hence the buffer of 1.
// For example, if we flush 1 batch/sec, then batching 2 batches/sec doesn't make us faster.
// This is our initial assumption, and there's potential in testing with higher numbers if there's a high variability in flush times - assuming we can preserve the order of the batches. For now, we'll stick to 1.
```

Reply:
nit: I don't think we need that many words convey meaning - in this case, I feel like less is more. I'm not fussed and you can pick (or not) any of my words apart.